addon/actions/action.js

```diff
@@ -32,6 +32,10 @@ export default EmberObject.extend({
   init() {
     this._super(...arguments);
     assert('Model has to be persisted!', !(this.get('instance') && !this.get('model.id')));
+
+    let payload = this.get('payload') || {};
+    assert('payload should be an object',  emberTypeOf(payload) === 'object');
```

Comment:
BTW - Shouldn't be there possibility to send primitives (string, numbers) or array through payload?

Reply:
nope... the request payload should be always as an Object. In REST and JSONAPI patterns